azure/activity_logs_monitoring/index.js

```diff
@@ -261,34 +257,43 @@ class EventhubLogHandler {
             var source = originalRecord['ddsource'];
             var config = this.logSplittingConfig[source];
             if (config !== undefined) {
-                var fields = config.path;
+                var splitFieldFound = false;
 
-                if (config.keep_original_log) {
-                    this.records.push(originalRecord);
-                }
-
-                var recordsToSplit = this.findSplitRecords(record, fields);
-                if (recordsToSplit === null) {
-                    return;
-                }
+                for (var i = 0; i < config.paths.length; i++) {
+                    var fields = config.paths[i];
+                    var recordsToSplit = this.findSplitRecords(record, fields);
+                    if (recordsToSplit === null) {
+                        continue;
+                    }
 
-                for (var j = 0; j < recordsToSplit.length; j++) {
-                    var splitRecord = recordsToSplit[j];
-                    if (typeof splitRecord === 'string') {
-                        try {
-                            splitRecord = JSON.parse(splitRecord);
-                        } catch (err) {
-                            splitRecord = { message: splitRecord };
+                    splitFieldFound = true;
+                    for (var j = 0; j < recordsToSplit.length; j++) {
+                        var splitRecord = recordsToSplit[j];
+                        if (typeof splitRecord === 'string') {
```

Comment:
is dictionary the other type splitRecord could be?

Reply:
Technically it could be anything - Thanks for asking. I added some type checks that should catch when it's not a map